File: 18주차/SG/Coin_Change.cpp
#include<bits/stdc++.h>
using namespace std;
#define MAX 10001
class Solution {
public:
    int coinChange(vector<int> coins, int amount) {
        vector<int> dp(MAX,99999);
        sort(coins.begin(),coins.end());
        dp[0] = 0;
        for(int i=1;i<=amount;i++){
            for(auto coin : coins){
                if(i-coin < 0) continue;
                dp[i] = min(dp[i-coin]+1,dp[i]);
            }
        }
        if(dp[amount] == 99999) return -1;
        return dp[amount];
    }
};

File: 18주차/YH/coinChange.py
class Solution:
    def coinChange(self, coins: List[int], amount: int) -> int:
        
        dp = [amount+1 for i in range(amount+1)]
        dp[0] = 0
        
        for i in range(amount+1) :
            for k in range(len(coins)) :
                if i-coins[k] >= 0 :
                    dp[i] = min(dp[i],dp[i-coins[k]]+1)
                    
        if dp[amount] == amount+1 : return -1
        else : return dp[amount]